src/v/kafka/protocol/schemata/generator.py

```diff
@@ -534,6 +559,12 @@ class ScalarType(FieldType):
     def __init__(self, name):
         super().__init__(name)
 
+    @property
+    def potentially_flexible_type(self):
+        """Evaluates to true if the scalar type would be parsed as flex
+        if the version is high enough"""
+        return self.name == "string" or self.name == "bytes" or self.name == "iobuf"
```

Comment:
why is fetch record set not in this list?

Reply:
This is only for scalar types